R/SouthAfrica.R

```diff
@@ -0,0 +1,85 @@
+#' SouthAfrica Class for downloading, cleaning and processing notification data
+#' @description Country specific information for downloading, cleaning
+#'  and processing covid-19 region data for South Africa.
+#'
+#' @details Inherits from `DataClass`
+#' @source https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_provincial_cumulative_timeline_confirmed.csv # nolint
+#' @source https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_provincial_cumulative_timeline_deaths.csv # nolint
+#' @export
+#' @examples
+#' \dontrun{
+#' region <- SouthAfrica$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+SouthAfrica <- R6::R6Class("SouthAfrica",
+  inherit = DataClass,
+  public = list(
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field country name of country to fetch data for
+    country = "South Africa",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list("1" = "province"),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list("1" = "iso_3166_2"),
+    #' @field common_data_urls List of named links to raw data. The first, and
+    #' only entry, is be named main.
+    common_data_urls = list(
+      "main" = "https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_provincial_cumulative_timeline_confirmed.csv" # nolint
+    ),
+    #' @field level_data_urls List of named links to raw data that are level
+    #' specific.
+    level_data_urls = list(
+      "1" = list("deaths" = "https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_provincial_cumulative_timeline_deaths.csv") # nolint
+    ),
```

Comment:
If we only have level 1 data for South Africa then these should both be in `common_data_urls`; `level_data_urls` can be omitted and (at least in my thinking) you could rename `main` to `cases` for clarity and readability. @seabbs may differ on the `main` question.

Reply:
Yeah that makes more sense